### PR TITLE
[CDU] Add check for destination data, improvements to Type II Messages

### LINKED ISSUE #2031: [REQUEST] Behavior APPR in PERF in FMGC
**Is your feature request related to a problem? Please describe.**
NO

**Describe the feature you'd like to see implemented**
Change the behavoir in FMGC -> PERF -> APP as described under the following link ("ENTER DEST DATA message appears in amber on the scratchpad when within 180 NM of the destination and data is not entered" and for the QNH "Beyond 180 NM of the destination, brackets display. Within 180 NM, amber boxes appear."):

**References**
(http://a320dp.com/A320_DP/nav-flight-management/sys-14.5.40.html)

**Additional context**
It#s a "nice 2have" feature

AND THANKS to all of you for this excellent work!

Discord username (if different from GitHub):

### PR DESCRIPTION
**Pull request by MisterChocker**
Resolves #2031

## Summary of Changes

- Added check for destination data according to the references.
- Fixed Type II Messages show bad scratchpad input on clear.
- Added validity check for Type II Messages.
_reactionary, will check message validity before displaying_
- Added auto clear function for Type II Messages. (to clear obsolete messages)
_reactionary, check message queue on new message_
- Made Type II Message System react only

## Screenshots (if necessary)

image

## References

image

## Additional context

Fixes #2031

Discord username (if different from GitHub): DerL30N#3751

## How to download the PR for QA

Every new commit to this PR will cause a new A32NX artifact to be created and uploaded.
The build script will have already been run with the latest changes, so no need to rerun it once you download the zip.

1. Make sure you are signed in to GitHub
1. Click on the **Checks** tab on the PR
1. On the right side, click on the **Artifacts** drop down and click the **A32NX** link

### CODE REVIEW

**Review by Benjozork — APPROVED** [review]
LGTM

**Comment by danbw9** [comment]
Category: QA Test
Name: danbw9#7028
Date of testing: Current date 19/11/2020
Version of the sim: 1.10.11.0
PR Tested: #2009 
Object of test: Checking out the added scratchpad message for - 'ENTER DEST DATA' in MCDU

Testing technique: Complete normal flight and look for expected behaviour

Steps you've made to test the Pull Request:

- Start up from C&D from LEBL to LSZH - normal flight but not entering any data into APP PERF page.
- At approximate 180 miles waiting for message. Pops up around 160 miles to go (estimate on ND) - but only after I initiated a DES in managed mode. 
- message keeps re appearing until all fields are filled out

Curious to see what other QA testers find out, otherwise I will check it out tomorrow, as this was a testing report after one flight. 

Issues (If Any): Small issue - not sure if directly related to this PR. Described below - v
Overall Rating: Good
Conclusions of the testing: Nice addition. I had a little issue with the scratchpad logic behaviour with when I am in the PERF page and the 'ENTER DEST DATA' scratchpad message is displayed. I clear it. Enter a value in scratchpad for any of QNH/TEMP/MAG WND. Put it into the appropriate field. 'ENTER DEST DATA' message appears again. Clear it. and the value I manually put in reappears. Tried it several times and had the same issue. I looked in docs and could not find an answer. The 180 miles to go to DEST - is it based on F-PLAN track miles or when directly from aircraft position to DEST airport is 180 miles? Great work as always @MisterChocker

**Comment by MisterChocker** [comment]
I fixed the bad scratchpad behavior on message clear, used the correct variable for distance to destination and I slightly increased the timeout between each check now 15 seconds.
By clearing the message, a new one will show up on top of your scratchpad inputs. (I found that a bit annoying)
By not clearing the message, it will be hidden by any input to the scratchpad (except clr) and maintains hidden until scratchpad is emtpy.

**Comment by MisterChocker** [comment]
Okay should be good to test again.

**Comment by InhumanReactions** [comment]
Category: QA Tester
Name: InhumanReactions#7129
Date of testing: Current date 21/11/20
Version of the sim: 1.10.11.0
PR Tested: #2009
Object of test: Add check for destination data

Testing technique: Description of how you tested the Pull Request
Steps you've made to test the Pull Request:
1. Spawned in C&D at ENVA for a full flight to EGLL.
2. Took off as usual. No problems encountered.
3. Started my descent and the ENTER DEST DATA came up as expected.
4. Cleared the message and it came back up as expected. 
5.Message automatically cleared when all APPR info was inputted into the MCDU.

Flight notes (If required): 

Issues (If Any): I didn't experience any issues at all. The ENTER DEST DATA came back even after I cleared it (Which I assume is supposed to happen) which is great. It went away after everything was filled in which was perfect.
Overall Rating: Good
Conclusions of the testing: This is a great PR. The message worked as expected and it is really useful to us pilots as it reminds you incase you forget. I didn't encounter any issues. Great work!

File: A32NX/SimObjects/AirPlanes/Asobo_A320_NEO/flight_model.cfg [hunk 1]
```diff
@@ -299,13 +299,13 @@ cruise_lift_scalar = 1.2
 parasite_drag_scalar = 1
 induced_drag_scalar = 1
 flap_induced_drag_scalar = 1
-elevator_effectiveness = 1
+elevator_effectiveness = 0.1
```
Comment (lukecologne) [inline A32NX/SimObjects/AirPlanes/Asobo_A320_NEO/flight_model.cfg:302]: This doesn't look like it belongs here

File: A32NX/SimObjects/AirPlanes/Asobo_A320_NEO/flight_model.cfg [hunk 2]
```diff
@@ -299,13 +299,13 @@ cruise_lift_scalar = 1.2
...
 elevator_maxangle_scalar = 1
 aileron_effectiveness = 1
 rudder_effectiveness = 1
 rudder_maxangle_scalar = 0.5
-pitch_stability = 0.25
-roll_stability = 0.25
+pitch_stability = 5.25
+roll_stability = 3.25
```
Comment (lukecologne) [inline A32NX/SimObjects/AirPlanes/Asobo_A320_NEO/flight_model.cfg:308]: Same here

**Review by lukecologne — CHANGES_REQUESTED** [review]

File: A32NX/SimObjects/AirPlanes/Asobo_A320_NEO/flight_model.cfg (hunk 1 above)
Comment (MisterChocker) [reply A32NX/SimObjects/AirPlanes/Asobo_A320_NEO/flight_model.cfg]: removed changes

File: A32NX/SimObjects/AirPlanes/Asobo_A320_NEO/flight_model.cfg (hunk 2 above)
Comment (MisterChocker) [reply A32NX/SimObjects/AirPlanes/Asobo_A320_NEO/flight_model.cfg]: removed changes

**Review by lukecologne — APPROVED** [review]

**Pull request merged**